### PR TITLE
[stdlib] Documentation revisions

### PR DESCRIPTION
**Pull request by natecook1000**
These revisions touch on a variety of areas, mostly focusing on recent changes to the stdlib.

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/FloatingPoint.swift.gyb [hunk 1]
```diff
@@ -1748,12 +2004,55 @@ extension BinaryFloatingPoint {
...
       exponentBitPattern: magnitudeOf.exponentBitPattern,
       significandBitPattern: magnitudeOf.significandBitPattern)
   }
 
+  /// Returns a Boolean value indicating whether this instance should precede the
+  /// given value in an ascending sort.
+  ///
+  /// This relation is a refinement of the less-than-or-equal-to operator
+  /// (`<=`) that provides a total order on all values of the type, including
+  /// noncanonical encodings, signed zeros, and NaNs. Because it is used much
```
Comment (xwu) [inline stdlib/public/core/FloatingPoint.swift.gyb]: Actually, this isn't quite right. Per the IEEE specification:

> NOTE—totalOrder does not impose a total ordering on all encodings in a format. In particular, it does not distinguish among different encodings of the same floating-point representation, as when one or both encodings are non-canonical.

File: stdlib/public/core/FloatingPoint.swift.gyb [hunk 2]
```diff
@@ -1748,12 +2004,55 @@ extension BinaryFloatingPoint {
...
+  /// less frequently than the usual comparisons, there is no operator form of
+  /// this relation.
```
Comment (xwu) [inline stdlib/public/core/FloatingPoint.swift.gyb]: I don't think justification is required; in general, we should need to justify why certain facilities *are* operators and not why others are not.

By including this statement, I worry that you're prompting people to wonder whether they actually want to use this operation more often than they thought, as the implication is that an operator spelling was a real possibility that was considered and rejected.

File: stdlib/public/core/FloatingPoint.swift.gyb (hunk 1 above)
Comment (xwu) [inline stdlib/public/core/FloatingPoint.swift.gyb]: How would you feel about actually documenting how the total ordering works here?

Comment (natecook1000) [inline stdlib/public/core/FloatingPoint.swift.gyb]: Hmmm. So I know that our implementation *does* distinguish between encodings, so what we're saying here that `isTotallyOrdered` imposes the IEEE total order relationship *and further* imposes a total order even on noncanonical encodings. 

Is the actual order mandated by the spec? If it's not I don't know that we want to guarantee a specific ordering of the various elements. It's not immediately apparent from the implementation what ends up where.

@stephentyrone?

File: stdlib/public/core/FloatingPoint.swift.gyb (hunk 2 above)
Comment (natecook1000) [inline stdlib/public/core/FloatingPoint.swift.gyb]: That's a good point. I think this language might have originated in the floating-point protocols proposal, where it made more sense. I'll look at removing it.

File: stdlib/public/core/FloatingPoint.swift.gyb (hunk 1 above)
Comment (xwu) [inline stdlib/public/core/FloatingPoint.swift.gyb]: The actual order is mandated by the specification in gory detail. Some encodings of the same value are distinguished from each other, but not all.

Comment (natecook1000) [inline stdlib/public/core/FloatingPoint.swift.gyb]: Thanks, got it! Updated that and then noticed that the example (a) had the wrong parameter label and (b) didn't have the correct semantics... 🙄 Going to come back to the Q about specifying the order later.

Comment (natecook1000) [inline stdlib/public/core/FloatingPoint.swift.gyb]: I actually wish that method was `static func areInIncreasingTotalOrder(_ x: Self, _ y: Self) -> Bool`—would play more nicely with sorting.

Comment (xwu) [inline stdlib/public/core/FloatingPoint.swift.gyb]: Hmm, I see your point, but IEEE `totalOrder` actually _can't_ be used as a Swift sorting predicate: it is required to return `true` if two values are equal, whereas `areInIncreasingOrder` should return `false`.

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by moiseev** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/StringIndexConversions.swift [hunk 3]
```diff
@@ -12,7 +12,13 @@
 
 extension String.Index {
   /// Creates an index in the given string that corresponds exactly to the
-  /// specified `UnicodeScalarView` position.
+  /// specified position.
+  ///
+  /// If the index passed as `sourcePosition` represents the start of an
+  /// extended grapheme cluster---the element type of a string---then the
+  /// initializer succeeds. If the index instead represents the position of a
+  /// Unicode scalar within an extended grapheme cluster or the position of an
```
Comment (milseman) [inline stdlib/public/core/StringIndexConversions.swift]: I don't think this working needs to have anything to do with Unicode scalars. The semantics are specifically that if the index lies on a grapheme boundary, then it is valid. Non-grapheme boundary indices (whether on scalar boundaries or sub-scalar indices) return nil.

**Review by milseman — COMMENTED** [review]
Overall, awesome!

File: stdlib/public/core/UnsafeRawBufferPointer.swift.gyb [hunk 4]
```diff
@@ -490,31 +504,33 @@ public struct Unsafe${Mutable}RawBufferPointer
...
-  ///   MemoryLayout<T>.stride]                 
-  ///   is bound to type `T`.
-  ///
-  /// - Postcondition: The `T` values at `self[startIndex..<source.count *
-  ///   MemoryLayout<T>.stride]`
-  ///   are initialized.
+  /// Initializes the buffer's memory with the given elements, binding the
+  /// initialized memory to the elements' type.
+  ///
+  /// When calling the `initialize(as:from:)` method on a buffer `b`, the
```
Comment (atrick) [inline stdlib/public/core/UnsafeRawBufferPointer.swift.gyb]: Do you mean `initializeMemory(as:from:)`?

**Review by atrick — COMMENTED** [review]
Thanks.

**Pull request merged**